### PR TITLE
Add Format function for `SetToNotifsDiff`

### PR DESCRIPTION
**Pull request by wenovus**

### CODE REVIEW

File: gnmidiff/set_to_get.go [hunk 1]
```diff
@@ -15,18 +15,32 @@ import (
...
 }
 
+// Format outputs the SetToNotifsDiff in human-readable format.
+//
+// NOTE: Do not depend on the output of this being stable.
+func (diff SetToNotifsDiff) Format(f Format) string {
+	f.title = "SetToNotifsDiff"
+	f.aName = "want/SetRequest"
+	f.bName = "got/Notifications"
+	return SetRequestIntentDiff{
```
Comment (DanG100) [inline gnmidiff/set_to_get.go:36]: i'd extract the common logic of Format into its own struct instead using the SetRequestIntentDiff struct

```
type StructuredDiff struct {} 
func (fd StructuredDiff) Format()

type SetToNotifsDiff struct { StructuredDiff } 
type SetRequestIntentDiff struct { StructuredDiff } 
```

Comment (wenovus) [reply gnmidiff/set_to_get.go]: Made more modular, PTAL

File: gnmidiff/diff.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+// by ygot.PathToString.
+// - The value of the update fields is the JSON_IETF representation of the
+// value. This is to facilitate comparing JSON_IETF-represented values whose
+// real data type is obscured without knowledge of the YANG schema.
+type StructuredDiff struct {
+	DeleteDiff
+	UpdateDiff
+}
+
+type DeleteDiff struct {
```
Comment (DanG100) [inline gnmidiff/diff.go:46]: godoc comment for this type

File: gnmidiff/diff.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+	MissingDeletes map[string]struct{}
+	ExtraDeletes   map[string]struct{}
+	CommonDeletes  map[string]struct{}
+}
+
+type UpdateDiff struct {
```
Comment (DanG100) [inline gnmidiff/diff.go:52]: godoc comment for the type

**Review by DanG100 — APPROVED** [review]

**Pull request merged**